### DISCUSSION TITLE
Specification for `plotBoxAndWhisker()`

### DISCUSSION THREAD

**Discussion by IndrajeetPatil** [Open-Systems-Pharmacology/OSPSuite-R#813] [Ideas]
Related to #674 

# Signature

The function is called `plotBoxAndWhisker()`, and not `plotBoxWhisker()` because the latter already exists in _tlf_ package, and we should avoid any namespace collision.

```r
# x The variable to be plotted against grouping variable.
# `...` will be parameters to control annotations and aesthetics of the plot
plotBoxAndWhisker(dataCombined, x, ...)
```

# Implementation plan

```r
data <- dataCombined$toDataFrame()

tlf::plotBoxWhisker(data, ...) 
```

# Rough expected output:

(The x-axis will be mapped to `group` variable in the dataframe.)

![image](https://user-images.githubusercontent.com/11330453/157647322-0eba310e-b63b-4849-a848-ea9ce229458d.png)

# Decision points:

N.B. The option already rejected by the team is scratched out.

- [ ] Should `plotBoxAndWhisker()` have `x` argument that allows the users to specify the variable versus should it always be a fixed variable like _residuals_?
- [ ] Should the values be plotted for simulated versus observed data? (e.g., if `x = yValues`, which `yValues` does it correspond to?)

# Tasks

Each task will get its own PR.

- [ ] Implement `plotBoxAndWhisker()`

**Comment by IndrajeetPatil** [comment]
@PavelBal Do we actually need to implement this plot? I do not see it mentioned in the [original issue](https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/674). I just assumed that it was needed.

Also, it's such a simple plot that nobody needs to go the trouble of going the route of an `R6` object. 

We are also not implementing DDI, PK ratio, etc. plots, so it won't be internally inconsistent to not have every plotting functions take in a `DataCombined` object.

**Comment by PavelBal** [comment]
@IndrajeetPatil Not now, low prio.
